Bottom: 37f668c4d16094d2c05ee4b25f991fbd5cc40bd7
Top:    69d15294161e589ef106168550b5028b765dda01
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-07 18:59:49 +0100

Refresh of attempt-to-refactor-the-step

---

diff --git a/src/hid/gtk/object3d.c b/src/hid/gtk/object3d.c
index 7ef93eb..a79d255 100644
--- a/src/hid/gtk/object3d.c
+++ b/src/hid/gtk/object3d.c
@@ -38,27 +38,6 @@
 
 static GList *object3d_test_objects = NULL;
 
-static void
-print_edge_id (edge_ref e)
-{
-  printf ("ID %i.%i", ID(e), (unsigned int)e & 3u);
-}
-
-static void
-debug_print_edge (edge_ref e, void *data)
-{
-  printf ("Edge ID %i.%i\n", ID(e), (int)e & 3u);
-
-  printf ("Edge ONEXT is "); print_edge_id (ONEXT(e)); printf ("\n");
-  printf ("Edge OPREV is "); print_edge_id (OPREV(e)); printf ("\n");
-  printf ("Edge DNEXT is "); print_edge_id (DNEXT(e)); printf ("\n");
-  printf ("Edge DPREV is "); print_edge_id (DPREV(e)); printf ("\n");
-  printf ("Edge RNEXT is "); print_edge_id (RNEXT(e)); printf ("\n");
-  printf ("Edge RPREV is "); print_edge_id (RPREV(e)); printf ("\n");
-  printf ("Edge LNEXT is "); print_edge_id (LNEXT(e)); printf ("\n");
-  printf ("Edge LPREV is "); print_edge_id (LPREV(e)); printf ("\n");
-}
-
 void
 object3d_test_init (void)
 {
@@ -417,24 +396,19 @@ object3d_export_to_step (object3d *object, char *filename)
       contour3d *contour = contour_iter->data;
       edge_ref edge;
       step_id edge_loop;
+      step_id_list edge_loop_edges = NULL;
 
-      edge_loop = step_edge_loop (step, "NONE", 
+      edge = contour->first_edge;
+      do {
+        edge_loop_edges = g_list_append (edge_loop_edges, GINT_TO_POINTER (ORIENTED_EDGE_IDENTIFIER (edge)));
+      } while (edge = LNEXT (edge), edge != contour->first_edge);
 
-      fprintf (f, "#%i = EDGE_LOOP ( 'NONE', ", step->next_id);
-
-      /* Emit the edges.. */
-      fprintf (f, "(");
-      for (edge = contour->first_edge;
-           edge != LPREV (contour->first_edge);
-           edge = LNEXT (edge)) {
-        fprintf (f, "#%i, ", ORIENTED_EDGE_IDENTIFIER(edge));
-      }
-      fprintf (f, "#%i)", ORIENTED_EDGE_IDENTIFIER(edge));
-      fprintf (f, " ) ; ");
+      edge_loop = step_edge_loop (step, "NONE", edge_loop_edges);
 
-      fprintf (f, "#%i = FACE_%sBOUND ( 'NONE', #%i, .T. ) ;\n", step->next_id + 1, outer_contour ? "OUTER_" : "", step->next_id);
-      contour->face_bound_identifier = step->next_id + 1;
-      step->next_id = step->next_id + 2;
+      if (outer_contour)
+        contour->face_bound_identifier = step_face_outer_bound (step, "NONE", edge_loop, true);
+      else
+        contour->face_bound_identifier = step_face_bound (step, "NONE", edge_loop, true);
 
       face_contour_list = g_list_append (face_contour_list, GINT_TO_POINTER (contour->face_bound_identifier));
     }
diff --git a/src/hid/gtk/object3d.h b/src/hid/gtk/object3d.h
index ecdc721..26b87fe 100644
--- a/src/hid/gtk/object3d.h
+++ b/src/hid/gtk/object3d.h
@@ -15,7 +15,6 @@ void object3d_set_appearance (object3d *object, appearance *appear);
 void object3d_add_edge (object3d *object, edge_ref edge);
 void object3d_add_vertex (object3d *object, vertex3d *vertex);
 void object3d_add_face (object3d *object, face3d *face);
-object3d *object3d_create_test_cube (void);
 GList *object3d_from_board_outline (void);
 void object3d_export_to_step (object3d *object, char *filename);
 void object3d_test_board_outline (void);
